src/main/java/com/deque/axe/android/AxeCalculatedPropsImpl.java

```diff
@@ -0,0 +1,216 @@
+
+    EditTextCalculatedProps(String text,
+                            String contentDescription,
+                            String labelText,
+                            String hint,
+                            String className) {
+      this.text = text;
+      this.contentDescription = contentDescription;
+      this.labelText = labelText;
+      this.hint = hint;
+      this.className = className;
+    }
+
+    @Override
+    public String calculatedNameProp() {
+      //TODO: Consider Text also.
+      if (!AxeTextUtils.isNullOrEmpty(contentDescription)
+              && !AxeTextUtils.isNullOrEmpty(hint)) {
+        return getPropAfterNullCheck(hint) + SPACE_HOLDER
+                  + getPropAfterNullCheck(labelText);
+      } else {
+        return getPropAfterNullCheck(contentDescription) + SPACE_HOLDER
+                  + getPropAfterNullCheck(hint) + SPACE_HOLDER
+                  + getPropAfterNullCheck(labelText);
+      }
+    }
+
+    @Override
+    public String calculatedRoleProp() {
+      return className;
+    }
+
+    @Override
+    public String calculatedValueProp() {
+      return text;
+    }
+
+    @Override
+    public String calculatedStateProp() {
+      return null;
+    }
+  }
+
+  static class ImageViewCalculatedProps implements AxeCalculatedProps {
+
+    private final String text;
+    private final String contentDescription;
+    private final String labelText;
+    private final String className;
+
+    ImageViewCalculatedProps(String text,
+                             String contentDescription,
+                             String labelText,
+                             String className) {
+      this.text = text;
+      this.contentDescription = contentDescription;
+      this.labelText = labelText;
+      this.className = className;
+    }
+
+    @Override
+    public String calculatedNameProp() {
+      return getPropAfterNullCheck(text) + SPACE_HOLDER
+              + getPropAfterNullCheck(contentDescription) + SPACE_HOLDER
+              + getPropAfterNullCheck(labelText);
+    }
+
+    @Override
+    public String calculatedRoleProp() {
+      return className;
+    }
+
+    @Override
+    public String calculatedValueProp() {
+      return null;
+    }
+
+    @Override
+    public String calculatedStateProp() {
+      return null;
+    }
+  }
+
+  static class TextViewCalculatedProps implements AxeCalculatedProps {
+
+    private final String text;
+    private final String contentDescription;
+    private final String className;
+
+    TextViewCalculatedProps(String text, String contentDescription, String className) {
+      this.text = text;
+      this.contentDescription = contentDescription;
+      this.className = className;
+    }
+
+    @Override
+    public String calculatedNameProp() {
+      return getPropAfterNullCheck(text) + SPACE_HOLDER
+              + getPropAfterNullCheck(contentDescription);
+    }
+
+    @Override
+    public String calculatedRoleProp() {
```

Comment:
There's a lot of redundancy in these overrides. Can AxeCalculatedProps function as a default interface and implement the most common cases?